gen/calc.py

```diff
@@ -547,6 +547,37 @@ def validate_exhibitor_storage_master_discovery(master_discovery, exhibitor_stor
             "`master_http_load_balancer` then exhibitor_storage_backend must not be static."
 
 
+def calculate_adminrouter_tls_version_override(adminrouter_tls_1_0_enabled, adminrouter_tls_1_1_enabled,
+                                               adminrouter_tls_1_2_enabled):
+    tls_versions = list()
+    if adminrouter_tls_1_0_enabled == 'true':
+        tls_versions.append('TLSv1')
+
+    if adminrouter_tls_1_1_enabled == 'true':
+        tls_versions.append('TLSv1.1')
+
+    if adminrouter_tls_1_2_enabled == 'true':
+        tls_versions.append('TLSv1.2')
+
+    tls_version_string = " ".join(tls_versions)
+    return tls_version_string
+
+
+def calculate_adminrouter_tls_cipher_override(adminrouter_tls_cipher_suite):
+    if adminrouter_tls_cipher_suite != '':
+        return 'true'
+    else:
+        return 'false'
+
+
+def validate_adminrouter_tls_version_override(adminrouter_tls_version_override, adminrouter_tls_1_0_enabled,
+                                              adminrouter_tls_1_1_enabled, adminrouter_tls_1_2_enabled):
+    if adminrouter_tls_version_override == '':
+        assert any([adminrouter_tls_1_0_enabled, adminrouter_tls_1_1_enabled, adminrouter_tls_1_2_enabled]) == 'false',\
```

Comment:
Lets break down this single line assert into variables to improve readability.

Pseudocode:
```
msg = (
  "When not explicitly setting a tls_version_override, "
   "at least one tls boolean (tls_1_0, tls_1_1, tls_1_2) must be set."
)
has_some_tls_option = any(... fix logic here ...)
assert has_some_tls_option, msg
```

Reply:
I'm happy to do this, but the syntax I used is consistent with other examples in calc.py - see line 541 (validate_exhibitor_storage_master_discovery). Please advise.